infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -303,42 +292,42 @@ private static void startServer(Map<String, Object> opts) {
      * - pick the correct logging level before outputting error messages
      * - add serverEndpoint information
      *
-     * @param opts command line parameters
-     * @throws JoranException logback exception
+     * @param conf Server Configuration parameters
      */
-    private static void configureLogger(Map<String, Object> opts) {
+    private static void configureLogger(ServerConfiguration conf) {
         final Logger root = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
-        final Level level = Level.toLevel(((String) opts.get("--log-level")).toUpperCase());
+        final Level level = conf.getLogLevel();
         root.setLevel(level);
     }
 
     /**
      * Create the service directory if it does not exist.
      *
-     * @param opts Server options map.
+     * @param conf Server Configurations.
      */
-    private static void createServiceDirectory(Map<String, Object> opts) {
-        if ((Boolean) opts.get("--memory")) {
+    private static void createServiceDirectory(ServerConfiguration conf) {
+        if (conf.isInMemoryMode()) {
             return;
         }
 
-        File serviceDir = new File((String) opts.get("--log-path"));
-
-        if (!serviceDir.isDirectory()) {
+        File parentDir = new File(conf.getServerDir());
+        if (!parentDir.isDirectory()) {
```

Comment:
`isFile()` ?

Reply:
Would be great if you'd use `Path` instead of file

Reply:
I don't think isFile() is appropriate here, because we only want to succeed if the parentDir exists and is a directory. If we used isFile() instead of !isDir(), we would succeed even if the dir does not exist.

Also, the Path class doesn't provide isDir() method to check the condition.